bundles/org.openhab.binding.wundergroundupdatereceiver/README.md

```diff
@@ -0,0 +1,115 @@
+weather stations based on the Fine Offset Electronics WH2600-IP - this is simple, otherwise you have to set up dns
+such that it resolves the above hostname to your server, without preventing the server from resolving the proper ip if
+you want to forward the request.
+
+The server thus listens at http(s)://<your-openHAB-server>:<openHAB-port>/weatherstation/updateweatherstation.php and the weather station
+needs to be pointed at this address. If you can't configure the weather station itself to submit to an alternate hostname you would need
+to set up a dns server that resolves rtupdate.wunderground.com to the IP-address of your server and provide as dns to the weather station
+does DHCP. Make sure not to use this dns server instance for any other DHCP clients. 
+
+The request is in itself simple to parse, so by redirecting it to your openhab server you can intercept the values and use them to
+control items in your home. E.g. use measured wind-speed to close an awning or turn on the sprinkler system after some time without rain.
+This binding allows you to mix and match products from various manufacturers that otherwise have a closed system.
+
+If you wish to pass the measurements on to rtupdate.wunderground.com, you can use a simple rule that triggers on the
+wundergroundupdatereceiver:wundergroundUpdateReceiver:<channel-id>:metadata#last-query-trigger to do so. It can also be used to
+submit the same measurements to multiple weather services via multiple rules.
+
+## Supported Things
+
+Any weather station that sends weather measurement updates to the wunderground.com update URLs is supported.
+It is easiest to use with stations that have a configurable target address, but can be made to
+work with any internet-connected weather station, that gets it's dns server via DHCP or where the DNS server can be
+set. 
+
+## Discovery
+
+The binding starts listening at the above-mentioned URI as soon as it is initialized. Any request with an unregistered
+stationId is recorded and if auto-discovery is enabled appers in the inbox, otherwise can be registered when a manual
+scan is initiated. For each request parameter a channel is generated, based on a list of known parameters from
+https://support.weather.com/s/article/PWS-Upload-Protocol?language=en_US and other observed parameters from various
+weather station. If you have a weather station that submits a parameter that is unknown in the current version of the
+binding please feel free to submit an issue to have it added.
+
+## Thing Configuration
+
+The only configurable value is the station id, which should match the one configured on the weather station. If you don't plan on submitting
+measurements to wunderground.com, it can be any unique non-empty string value.
+
+## Channels
+
+Each measurement type the wunderground.com update service accepts has a channel. The channels are named the same as the request parameters they receive.
+Additionally there is a receipt timestamp and a trigger channel.
```

Comment:
The list of channels is dynamic. There are channel-types that are mapped to channels, but no specific list of channels is appropriate as it depends on the weather-station. Just the 2 versions of the weather-station software on the same hardware, that @MikeTheTux and I have used submit different lists of values.

Reply:
I understood that the channel name (and channel amount) is dynamic, but there's a finite list of channel types. The readme should also cover textual configuration. You could add a table of the channel types with name, item type and label.

Reply:
Unknown channels are added as generic channels configured with a itemtype that is guessed from the content, which the user can then reconfigure. As I understand it using a thing file prevents discovery and using the UI to configure items.

Iy would be useful to be able to export configuration to thing and item files in some way.

Reply:
It's okay if unknown channels can be added manually by the user. If I understand correctly the available channels depend on the weather station model. So, it would be helpful to list these channels in a table per model. Listing only the channels for *your* model would be sufficient. Other users could add the channels for their models, then.

> As I understand it using a thing file prevents discovery and using the UI to configure items.

The word "prevents" might be a bit misleading. It's up to the user to decide whether to use text files for the configuration or use the UI and discovery.

> Iy would be useful to be able to export configuration to thing and item files in some way.

I have the feeling there's some misconception. Can you elaborate what you mean by that?

Reply:
The thing is not the device it is the request being submitted to what the device(s) believe is wunderground.com. The current list of request parameters is derived from the WU documentation and various extras that people who have various devices have mentioned. Fx. @MikeTheTux wrote that one version of his devices firmware submitted a lowbat request parameter, while the net didn't.
I have added some more text to readme regarding this, as their seems to be some misconception about what the channels encompass.

Reply:
With regard to the "export from jsondb to .things/.items/etc files" functinality, the goal would to be enable backing up GUI generated configuration in a readily re-usable form, as someone mentioned in this PR. I have also read quite a few threads on the forums and other places where people miss this.
The challenge in manual configuration with this binding is that you have to intercept a request from each device submitting to WU, to see what to name the channels you configure, so they are recognised by the binding.